docs/internal/GeneralArchitectureGuide.md

```diff
@@ -68,6 +68,43 @@ state must ever be reloaded from persisted state.
 
 ## Deprecations
 
+## Backwards Compatibility
+
+Elasticsearch has major versions (e.g. 8.0.0), minor versions (e.g. 8.3.0) and patch versions (e.g. 8.3.1). Multiversion
+binary backwards compatibility within a cluster is guaranteed for specific situations.
```

Comment:
What exactly do you mean by "binary" here? Could we drop it here (and below) without losing anything?